client/reader/user-profile/components/user-profile-header/test/index.tsx

```diff
@@ -0,0 +1,213 @@
+		render( <UserProfileHeader user={ defaultUser } /> );
+
+		// Check if navigation section is rendered
+		// @ts-expect-error -- jest-dom matchers are available globally
+		expect( screen.getByTestId( 'section-nav' ) ).toBeInTheDocument();
+		// @ts-expect-error -- jest-dom matchers are available globally
+		expect( screen.getByTestId( 'nav-tabs' ) ).toBeInTheDocument();
+
+		// Check for navigation items
+		const navItems = screen.getAllByTestId( 'nav-item' );
+		expect( navItems.length ).toBe( 2 ); // Posts and Lists
+
+		// Check nav item content - should have Posts and Lists
+		const navTexts = navItems.map( ( item ) => item.textContent );
+		expect( navTexts ).toContain( 'Posts' );
+		expect( navTexts ).toContain( 'Lists' );
+	} );
+
+	/**
+	 * Test: Bio not rendered when not provided
+	 *
+	 * Verifies that:
+	 * 1. Bio section is not rendered when user has no bio
+	 */
+	test( 'should not render bio section when user has no bio', () => {
+		render( <UserProfileHeader user={ defaultUser } /> );
+
+		// Bio section should not be present
+		const bioSection = document.querySelector( '.user-profile-header__bio' );
+		// @ts-expect-error -- jest-dom matchers are available globally
+		expect( bioSection ).not.toBeInTheDocument();
+	} );
+
+	/**
+	 * Test: Bio rendering
+	 *
+	 * Verifies that:
+	 * 1. Bio section is rendered when user has a bio
+	 * 2. The bio text is displayed correctly
+	 */
+	test( 'should render bio section when user has a bio', () => {
+		const userWithBio = {
+			...defaultUser,
+			bio: 'This is my test biography that describes me as a test user.',
+		};
+
+		render( <UserProfileHeader user={ userWithBio } /> );
+
+		// Bio section should be present
+		const bioText = screen.getByText( userWithBio.bio );
+		// @ts-expect-error -- jest-dom matchers are available globally
+		expect( bioText ).toBeInTheDocument();
+		// @ts-expect-error -- jest-dom matchers are available globally
+		expect( bioText.closest( '[class*="user-profile-header__bio"]' ) ).toBeInTheDocument();
+	} );
+
+	/**
+	 * Test: Bio with profile URL
+	 *
+	 * Verifies that:
+	 * 1. Read More link is displayed when bio is long and profile URL exists
+	 */
+	test( 'should render Read More link for long bio with profile URL', () => {
+		// Create a long bio that will likely be clamped
+		const longBio =
+			'This is a very long biography that will definitely exceed the three lines limit and therefore should display the Read More link. '.repeat(
+				5
+			);
+		const userWithLongBio = {
+			...defaultUser,
+			bio: longBio,
+		};
+
+		// We need to mock Element.scrollHeight and Element.offsetHeight to simulate text being clamped
+		Object.defineProperty( HTMLElement.prototype, 'scrollHeight', {
+			configurable: true,
+			get: function () {
+				// Return a large value for bio text element to simulate clamping
+				if ( this.classList.contains( 'user-profile-header__bio-desc-text' ) ) {
+					return 200;
+				}
+				return 100;
+			},
+		} );
```

Comment:
Can we mock the `isClamped` state instead of this?

Reply:
I took a stab at this but couldn't get it to work. I propose we stick to this implementation unless you have better luck. 😅

Reply:
Tested but cannot make it work. The current implementation is having a global effect so can we please avoid it? Seems like we can also remove `configurable` param from the function.

Reply:
Gotcha. I've removed the "long bio read more" test since that implementation is problematic, and I don't have a reliable alternative. 835d2877c7fcb91ab3dc519b3c6d99d3991a5202